File: problems/2760_longest_even_odd_subarray_with_threshold/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int longestAlternatingSubarray(vector<int>& nums, int threshold) {
        int ans = 0, i = 0, n = nums.size();
        while (i < n) {
            if (nums[i]%2==0 && nums[i]<=threshold) {
                int j = i;
                while (j+1<n && nums[j+1]%2!=nums[j]%2 && nums[j+1]<=threshold) j++;
                ans = max(ans, j-i+1); i = j+1;
            } else i++;
        }
        return ans;
    }
};

File: problems/2760_longest_even_odd_subarray_with_threshold/solution.py
from typing import List


class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        ans = 0
        n = len(nums)
        i = 0
        while i < n:
            if nums[i] % 2 == 0 and nums[i] <= threshold:
                j = i
                while j + 1 < n and nums[j+1] % 2 != nums[j] % 2 and nums[j+1] <= threshold:
                    j += 1
                ans = max(ans, j - i + 1)
                i = j + 1
            else:
                i += 1
        return ans
